pinot-core/src/main/java/org/apache/pinot/core/data/manager/offline/DimensionTableDataManager.java

```diff
@@ -161,4 +168,8 @@ public GenericRow lookupRowByPrimaryKey(PrimaryKey pk) {
   public FieldSpec getColumnFieldSpec(String columnName) {
     return _tableSchema.getFieldSpecFor(columnName);
   }
+
+  public List<String> getPrimaryKeyColumns() {
+    return new ArrayList<>(_primaryKeyColumns);
```

Comment:
(nit) no need to make a copy of the array (save the garbage). We don't usually guard the callers from modifying the list as it is internal code.

Reply:
Removed.